examples/exampleapp/exampleapp_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) The EfficientGo Authors.
+// Licensed under the Apache License 2.0.
+
+package exampleapp
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/efficientgo/e2e"
+	e2edb "github.com/efficientgo/e2e/db"
+	e2einteractive "github.com/efficientgo/e2e/interactive"
+	"github.com/efficientgo/tools/core/pkg/backoff"
+	"github.com/efficientgo/tools/core/pkg/testutil"
+)
+
+func TestExampleApp(t *testing.T) {
+	// Start a new Docker environment.
+	e, err := e2e.NewDockerEnvironment("e2e_example_app")
+	testutil.Ok(t, err)
+	// Make sure resources are cleaned up.
+	t.Cleanup(e.Close)
+
+	fmt.Println("=== Start example application...")
+	app := e2e.NewInstrumentedRunnable(e, "example_app").
+		WithPorts(map[string]int{"http": 8080}, "http").
+		Init(e2e.StartOptions{
+			Image: "quay.io/brancz/prometheus-example-app:v0.3.0",
+		})
+	testutil.Ok(t, e2e.StartAndWaitReady(app))
+
+	config := fmt.Sprintf(`
+global:
+  external_labels:
+    prometheus: prometheus-example-app
+scrape_configs:
+- job_name: 'example-app'
+  scrape_interval: 1s
+  scrape_timeout: 1s
+  static_configs:
+  - targets: [%s]
+  relabel_configs:
+  - source_labels: ['__address__']
+    regex: '^.+:80$'
+    action: drop
+`, app.InternalEndpoint("http"))
+
+	fmt.Println("=== Start Prometheus")
+	// Create Prometheus instance and wait for it to be ready.
+	p1 := e2edb.NewPrometheus(e, "prometheus-1")
+	testutil.Ok(t, p1.SetConfig(config))
+	testutil.Ok(t, e2e.StartAndWaitReady(p1))
+
+	fmt.Println("=== Ensure that Prometheus already scraped something")
+	// Ensure that Prometheus already scraped something.
+	testutil.Ok(t, p1.WaitSumMetrics(e2e.Greater(50), "prometheus_tsdb_head_samples_appended_total"))
+
+	// Open example in browser.
```

Comment:
Can we automate this and send some HTTP requests from the test itself? If more than 5 requests are needed for further manual testing purposes we could then hit the URL a few more times.

Reply:
ah fair point - I think still would be nice to show manually how this can work with doing the requests manually in the browser @matej-g what do you think?

Reply:
@douglascamata is there any reason you'd want this to be automatic specifically? As @jessicalins, we intentionally want to wait on user's input, to showcase the capabilities of the framework in a more interactive way.

Reply:
To allow people to get quicker to the last stage of the interaction (seeing the output) if they desire so. Doing the minimum amount of requests ensure this step can be optionally skipped. But more requests can be done manually while watching the output change.

Reply:
Imagine if every single time you run this test you have to open the browser and refresh a page 5 times. Might get boring.

Reply:
I think you'd normally not run this test multiple times, since it's a simple example to showcase some capabilities of the framework. With automatic requests, the test scenario will progress without user's interaction - but we deliberately want to wait on user's input in this case, so that the test progression is not 'hidden' from user. The browser is opened automatically for user and it takes 2-3 refreshes to make the test progress (since browser does favico request as well on each page load, 5 requests can be reached fast).